src/ekf.cpp

```diff
@@ -362,7 +362,17 @@ namespace RobotLocalization
     estimateErrorCovariance_ = (transferFunctionJacobian_ *
                                 estimateErrorCovariance_ *
                                 transferFunctionJacobian_.transpose());
-    estimateErrorCovariance_.noalias() += (processNoiseCovariance_ * delta);
+
+    Eigen::MatrixXd processNoise = processNoiseCovariance_;
+    // If the velocity is zero set the position process noise to zero
+    // This make it so that the covariance will not grow if the robot is not moving
+    if (xVel*xVel + yVel*yVel + zVel*zVel < zeroVelocityThreshold_
+        || pitchVel*pitchVel + rollVel*rollVel + yawVel*yawVel < zeroVelocityThreshold_)
```

Comment:
I think you want the total magnitude of the velocity not each component. This is particularly true if we are using an omnidirectional platform like the ridgeback.